getinsomnia/insomnia
# feat(pre-req): enable manipulation of proxy and certificate through insomnia.request - INS-3379

### Please review this PR (https://github.com/Kong/insomnia/pull/7170) firstly as current one is based on it
This is change is not part of request manipulation PR because this basically manipulate values from settings and clientCertificates, and for making PRs small.

Changes:
- [x] enabled manipulation certificate and proxy through `insomnia.request`
- [x] added smoke tests
- [x] added UTs


### How to test
Run this script and check if it runs well:

```javascript
// print current proxy and modify it
console.log(insomnia.request.proxy.getProxyUrl());
insomnia.request.proxy.update({
        host: 'localhost',
        match: '<all_urls>',
        port: 8888,
        tunnel: false,
        authenticate: false,
        username: '',
        password: '',
});

// print current certificate and modify it
// print the original one
console.log('key:', insomnia.request.certificate.key.src);
console.log('cert:', insomnia.request.certificate.cert.src);
console.log('passphrass:', insomnia.request.certificate.passphrass);
console.log('pfx:', insomnia.request.certificate.pfx.src);
// update
insomnia.request.certificate.update({
    disabled: true,
    key: {src: 'invalid.key'},
    cert: {src: 'invalid.cert'},
    passphrase: '',
    pfx: {src: ''},
});
```
